app/controllers/teams_controller.rb

```diff
@@ -74,6 +74,7 @@ def set_users
 
     def team_params
       params[:team].fetch(:sources_attributes, {}).delete_if { |key, source| source[:url].empty? }
+      params[:team].fetch(:roles_attributes, {}).delete_if { |key, role| role.present? && role[:github_handle].empty? }
```

Comment:
I'm not really sure what this line and the one above do. That is, I don't know if `#fetch` returns a _copy_ of the nested hash or the actual instance. Depending on what object it returns, deleting keys from it probably doesn't do anything.

In any event, I would remove the line again. The problem (or well: at least _one_ of the problems 😉 ) should be solved by your change in the github_handle concern.